src/test/ui/const-generics/parser-error-recovery/issue-89013-type.stderr

```diff
@@ -0,0 +1,24 @@
+error: missing type to the right of `=`
+  --> $DIR/issue-89013-type.rs:9:13
+   |
+LL | impl Foo<N = type 3> for Bar {
+   |             ^
+   |
+help: to constrain the associated type, add a type after `=`
+   |
+LL | impl Foo<N = TheType type 3> for Bar {
+   |              +++++++
+help: remove the `=` if `N` is a type
+   |
+LL - impl Foo<N = type 3> for Bar {
+LL + impl Foo<N type 3> for Bar {
+   | 
+
+error: expected one of `,`, `>`, a const expression, lifetime, or type, found keyword `type`
+  --> $DIR/issue-89013-type.rs:9:14
+   |
+LL | impl Foo<N = type 3> for Bar {
+   |              ^^^^ expected one of `,`, `>`, a const expression, lifetime, or type
```

Comment:
These look terrible! 😅

We might have to treat them exclusively 🤔

I think if we encounter keywords anywhere we don't recognize, we should likely bail and swallow the whole generic params list.

Reply:
Here's a new version of the error message, with the code tweaked to return an error in this case instead of just `Ok(None)`.

```
error: missing type to the right of `=`
  --> $DIR/issue-89013-type.rs:9:13
   |
LL | impl Foo<N = type 3> for Bar {
   |             ^---- expected type, found keyword `type`

error: aborting due to previous error
```